File: pset1/cash-credit/credit.c
#include <stdio.h>
#include <cs50.h>
#include <stdlib.h>
#include <math.h>

int main(void)
{
    long int number = 0;
    int sum[32];
    int digit[16];
    int total = 0;
    long int temp = 0;
    int temptotal = 0;
    int nDigits = 0;

    for (int i = 0; i < 35; i++)
    {
        sum[i] = 0;
        digit[i] = 0;
    }

    do
    {
        number = get_long("Credit card number:");
        nDigits = floor(log10(labs(number))) + 1 ;
        if (nDigits < 13)
        {
            printf("INVALID\n");
            return 0;
        }
    }
    while (nDigits < 13);
    temp = number;
    for (int i = 0; i < 17; i++)
    {
        digit[i] = temp % 10;
        temp = temp / 10;
    }
    for (int i = 1; i < 17; i += 2)
    {
        if (digit[i] * 2 >= 10)
        {
            sum[i - 1] = (digit[i] * 2) % 10;
            sum[i] = (digit[i] * 2 / 10);
        }
        else
        {
            sum[i - 1] = digit[i] * 2;
        }
    }
    for (int i = 0; i < 32; i++)
    {
        temptotal += sum[i];
    }
    for (int i = 0; i < 16; i += 2)
    {
        total += digit[i];
    }
    total = total + temptotal;
    if (total % 10 == 0)
    {
        if ((digit[14] != 0) && (digit[14] == 3) && (digit[13] == 4 || digit[13] == 7))
        {
            printf("AMEX\n");
        }
        else if ((digit[15] != 0) && (digit[15] == 5) && ((digit[14] == 1) || (digit[14] == 2) || (digit[14] == 3) || (digit[14] == 4)
                 || (digit[14] == 5)))
        {
            printf("MASTERCARD\n");
        }
        else if ((digit[12] == 4 || digit[13] == 4 || digit[14] == 4) || digit[15] == 4)
        {
            printf("VISA\n");
        }
        else
        {
            printf("INVALID\n");
        }
    }
    else
    {
        printf("INVALID\n");
    }
}

File: pset6/credit/credit.py
from cs50 import get_string
import math
import sys

# Prompting user for input.
while True:
    number = get_string("Credit card number:")
    nDigits = math.floor(math.log10(abs(int(number)))) + 1
    if nDigits < 13:
        print("INVALID")
        sys.exit(1)
    if nDigits >= 13:
        break

thesum = 0
temp = 0
temp_one = 0
temp_two = 0

# Checking for total sum. We do same math magic to get numbers to digits

for i in range(0, len(number), 1):
    if len(number) % 2 == 0:
        if i % 2 == 0:
            temp = (int(number[i])) * 2
            if temp > 9:
                temp_one = int(math.floor(temp / 10))
                temp_two = int(((temp * 10) - 100) / 10)
                thesum += temp_one
                thesum += temp_two
            else:
                thesum += temp
        else:
            thesum += int(number[i])
    else:
        if i % 2 == 0:
            thesum += int(number[i])
        else:
            temp = (int(number[i])) * 2
            if temp > 9:
                temp_one = int(math.floor(temp / 10))
                temp_two = int(((temp * 10) - 100) / 10)
                thesum += temp_one
                thesum += temp_two
            else:
                thesum += temp
if thesum % 10 == 0:
    if len(number) == 15 and int(number[0]) == 3 and (int(number[1]) == 4 or int(number[1]) == 7):
        print("AMEX")
    elif len(number) == 16 and int(number[0]) == 5 and (int(number[1]) == 1 or int(number[1]) == 2 or int(number[1]) == 3 or int(number[1]) == 4 or int(number[1]) == 5):
        print("MASTERCARD")
    elif (len(number) == 13 or len(number) == 16) and int(number[0]) == 4:
        print("VISA")

else:
    print("INVALID")